[BUG] Manually removed subaccount entitlements cause plugin crash

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What version of Terraform are you using?

1.5.6

### What type of issue are you facing

bug report

### Describe the bug

- Create subaccount and assign entitlements using the plugin
- log into BTP portal and manually delete the assignment
- `terraform plan` -> 💥  

```
Stack trace from the terraform-provider-btp_v0.3.0-beta1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1715081]

goroutine 228 [running]:
github.com/SAP/terraform-provider-btp/internal/provider.(*subaccountEntitlementResource).Read(0xc000392030, {0x1a1d1a0, 0xc000170a20}, {{{{0x1a217f0, 0xc00026f740}, {0x17e27e0, 0xc00026f530}}, {0x1a23e60, 0xc000624550}}, 0xc0003aa000, ...}, ...)
	github.com/SAP/terraform-provider-btp/internal/provider/resource_subaccount_entitlement.go:155 +0x421
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc0001cf4a0, {0x1a1d1a0, 0xc000170a20}, 0xc000170ab0, 0xc000457648)
	github.com/hashicorp/terraform-plugin-framework@v1.3.5/internal/fwserver/server_readresource.go:101 +0x677
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0xc0001cf4a0, {0x1a1d1a0?, 0xc0001708d0?}, 0xc000726280)
	github.com/hashicorp/terraform-plugin-framework@v1.3.5/internal/proto6server/server_readresource.go:55 +0x41a
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0xc0002a45a0, {0x1a1d1a0?, 0xc000170000?}, 0xc00054e240)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov6/tf6server/server.go:749 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x18a7f20?, 0xc0002a45a0}, {0x1a1d1a0, 0xc000170000}, 0xc0000c6d90, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:386 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00024a1e0, {0x1a22198, 0xc000007a00}, 0xc000460480, 0xc00039a150, 0x1ee5fb0, 0x0)
	google.golang.org/grpc@v1.56.1/server.go:1337 +0xdf3
google.golang.org/grpc.(*Server).handleStream(0xc00024a1e0, {0x1a22198, 0xc000007a00}, 0xc000460480, 0x0)
	google.golang.org/grpc@v1.56.1/server.go:1714 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.56.1/server.go:959 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.56.1/server.go:957 +0x18c

Error: The terraform-provider-btp_v0.3.0-beta1 plugin crashed!
```


### Expected Behavior

Entitlements are added back to match desired state

### Steps To Reproduce

_No response_

### Add screenshots to help explain your problem

_No response_

### Additional context

btp provider version `0.3.0-beta1`

--- a/internal/provider/resource_subaccount_entitlement.go
+++ b/internal/provider/resource_subaccount_entitlement.go
@@ -152,8 +152,12 @@ func (rs *subaccountEntitlementResource) Read(ctx context.Context, req resource.
 		return
 	}
 
-	updatedState, diags := subaccountEntitlementValueFrom(ctx, *entitlement)
+	if entitlement == nil {
+		resp.Diagnostics.AddError("API Error Reading Resource Entitlement (Subaccount)", "Resource not found")
+		return
+	}
 
+	updatedState, diags := subaccountEntitlementValueFrom(ctx, *entitlement)
 	resp.Diagnostics.Append(diags...)
 
 	diags = resp.State.Set(ctx, &updatedState)

